apps/website/src/examples/Dialog.fullpage.tsx

```diff
@@ -0,0 +1,44 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+import {
+  Modal,
+  Button,
+  ModalContent,
+  ModalButtonBar,
+  LabeledInput,
+  LabeledTextarea,
+} from '@itwin/itwinui-react';
+
+export default () => {
+  const [isModalOpen, setIsModalOpen] = React.useState(true);
+  const closeModal = () => setIsModalOpen(false);
+
+  return (
+    <>
+      <Button styleType='high-visibility' onClick={() => setIsModalOpen(true)}>
+        Open full page dialog
+      </Button>
+      <Modal
+        isOpen={isModalOpen}
+        title={'New message'}
+        styleType='fullPage'
+        onClose={() => closeModal()}
+        setFocus={false}
```

Comment:
Don't know if it is just for me for `setFocus` does not seem to exist on `ModalProps`. So VSCode shows an error when passing false to it.

Same in `Dialog.modal.tsx`

Reply:
Although TS is not allowing to pass `setFocus`, it is being applied using `{...rest}` which is why it does not focus which is good.

Any reason why we did not pick `setFocus` from `DialogMainProps`?

https://github.com/iTwin/iTwinUI/blob/090c60a2a4387e09a7967e471a63a8dbbd4c351d/packages/itwinui-react/src/core/Modal/Modal.tsx#L54

Can we pick `setFocus` and set the default to `true` for `Modal`?

Reply:
I don't remember why we did not include setFocus. 
We have it added in Modal.tsx line 110 which somehow works.

Reply:
> We have it added in Modal.tsx line 110 which somehow works.

I believe adding `setFocus` on [L110](https://github.com/iTwin/iTwinUI/blob/main/packages/itwinui-react/src/core/Modal/Modal.tsx#L110) of Modal.tsx is the shorthand to assign the value of `true` to a boolean prop. So the below two are the same:

```tsx
<>
<Dialog setFocus/>
<Dialog setFocus={true}/>
</>
```

So I guess should we allow to pass a `setFocus` in Modal and pass it down to the Dialog? The default value of the `setFocus` on Modal can be true to keep the same behavior.

Reply:
i believe this was done because Modal should always intercept and trap focus, unlike Dialog which can be non-modal.